sql/analyzer/filters.go

```diff
@@ -0,0 +1,84 @@
+package analyzer
+
+import (
+	"reflect"
+
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+	"gopkg.in/src-d/go-mysql-server.v0/sql/expression"
+)
+
+type filters map[string][]sql.Expression
+
+func (f filters) merge(f2 filters) {
+	for k, exprs := range f2 {
+		f[k] = append(f[k], exprs...)
+	}
+}
+
+func exprToTableFilters(expr sql.Expression) filters {
+	filtersByTable := make(filters)
+	for _, expr := range splitExpression(expr) {
+		var tables []string
+		_, _ = expr.TransformUp(func(e sql.Expression) (sql.Expression, error) {
+			f, ok := e.(*expression.GetField)
+			if ok {
+				tables = append(tables, f.Table())
+			}
+
+			return e, nil
+		})
+
+		if len(tables) == 1 {
+			filtersByTable[tables[0]] = append(filtersByTable[tables[0]], expr)
```

Comment:
I believe I don't understand this logic. Does it mean that the expression is only added to the left table?

Reply:
A filter is only pushed down if only one table is mentioned in the expression (unless we want to change that behavior)

Reply:
:+1: